### ISSUE TITLE
Make final version 4.0.0

### ISSUE THREAD

**Issue by mlojewski-me** [nextcloud/user_sql#41]
Make final version 4.0.0 in master branch and publish it to app store. The version should support Nextcloud 14.

- [x] Merge branch develop-14 into develop
- [x] Remove branch develop-14
- [x] Merge branch develop into master (change version to 4.0.0)
- [x] Publish app in Nextcloud store
- [x] Change version in develop to 4.1.0-dev

**Comment by mlojewski-me** [comment]
Hey,
I'm planning to make a release of new version along with release of Nextcloud 14. Some functions which current version uses are going to be dropped in the Nextcloud 14, so version 3.2.0 will not work on that.

I've changed the whole implementation, trying to write clean code. I also added some new functionalities. Unfortunately I haven't got time to write upgrade script, also I haven't implemented domain support.

Any feedback is welcomed.
